### PR TITLE
Add build invocation ID

### PR DESCRIPTION
**Pull request by ianlewis**
Updates #8

### CODE REVIEW

File: slsa/provenance.go [hunk 1]
```diff
@@ -58,7 +64,8 @@ func HostedActionsProvenance(w WorkflowRun) (*intoto.ProvenanceStatement, error)
 			Materials:   w.Materials,
 			// TODO(https://github.com/slsa-framework/slsa-github-generator/issues/8): support more metadata fields.
 			Metadata: &slsa.ProvenanceMetadata{
-				Completeness: w.Completeness,
+				BuildInvocationID: buildInvocationID,
+				Completeness:      w.Completeness,
```
Comment (asraa) [inline slsa/provenance.go:68]: i'm just curious about this, but is a workflowrun materials really completeness true (from `NewWorkflowRun`)?

For generic generator, I'm not sure

File: slsa/provenance.go [hunk 2]
```diff
@@ -42,6 +42,12 @@ func HostedActionsProvenance(w WorkflowRun) (*intoto.ProvenanceStatement, error)
 		builderID = fmt.Sprintf("https://github.com/%s", t.JobWorkflowRef)
 	}
 
+	buildInvocationID := w.GithubContext.RunID
+	if w.GithubContext.RunAttempt != "" {
+		// NOTE: RunID does not get updated on re-runs so we need to include RunAttempt.
+		buildInvocationID = fmt.Sprintf("%s-%s", w.GithubContext.RunID, w.GithubContext.RunAttempt)
```
Comment (asraa) [inline slsa/provenance.go:48]: Could you please add some documentation on the format of this field? 

(Maybe add here https://github.com/slsa-framework/slsa-github-generator#provenance-example and add a note)

**Review by asraa — COMMENTED** [review]
Thanks!

File: slsa/provenance.go (hunk 1 above)
Comment (ianlewis) [reply slsa/provenance.go:68]: Completeness gets taken from the `WorkflowRun`. The generic generator only supplies `completeness.parameters = true` based on a suggestion from @MarkLodato. All the rest of the fields are false.

https://github.com/slsa-framework/slsa-github-generator/blob/main/slsa/github.go#L117-L119

See https://github.com/slsa-framework/slsa-github-generator/pull/3#discussion_r840747519 for context when it was added.

File: slsa/provenance.go (hunk 2 above)
Comment (ianlewis) [reply slsa/provenance.go:48]: I will add something related to the invocation ID. I also added an issue to better document the fields determined by the `buildType` where we have made some judgements about the data there. 
https://github.com/slsa-framework/slsa-github-generator/issues/33

Comment (ianlewis) [reply slsa/provenance.go:48]: Done.

File: slsa/provenance.go (hunk 1 above)
Comment (ianlewis) [reply slsa/provenance.go:68]: I'm going to resolve as I think there isn't anything to do here but feel free to reopen if necessary.

**Review by laurentsimon — APPROVED** [review]

**Comment by laurentsimon** [comment]
Ready to merge?

**Comment by ianlewis** [comment]
> Ready to merge?

Yep. Just waiting for an approval.

**Pull request merged**